### PR TITLE
glslc: expose HLSL compilation support from Glslang

### PR DESCRIPTION
**Pull request by dneto0**
Support "-x hlsl", i.e. hlsl is a valid language specifier.

Files with .hlsl extension default to HLSL.

### CODE REVIEW

File: glslc/README.asciidoc [hunk 1]
```diff
@@ -12,6 +12,7 @@
 ----
 glslc [-c|-S|-E]
       [-x glsl] [-std=standard]
+      [-x hlsl]
```
Comment (antiagainst) [inline glslc/README.asciidoc]: I think we should write something like `[-x ...]` here to be consistent with the following.

BTW, `-std` should mention that it is not applicable to `-x hlsl` (if this is correct).

File: glslc/README.asciidoc [hunk 2]
```diff
@@ -189,8 +190,9 @@ under Vulkan semantics.
 
 ==== `-x`
 
-`-x` lets you specify the language of the input shader files. Right now, the
-only accepted argument is `glsl`.
+`-x` lets you specify the language of the input shader files. Valid languages
+are 'glsl' and 'hlsl'.  If the file extension is `hlsl` then the default language
```
Comment (antiagainst) [inline glslc/README.asciidoc]: formatting: backtick for `glsl` and `hlsl`

File: glslc/src/main.cc [hunk 3]
```diff
@@ -329,15 +347,22 @@ int main(int argc, char** argv) {
...
       // If current_fshader_stage is shaderc_glsl_infer_from_source, that means
       // we didn't set forced shader kinds (otherwise an error should have
       // already been emitted before). So we should deduce the shader kind
       // from the file name. If current_fshader_stage is specifed to one of
       // the forced shader kinds, use that for the following compilation.
-      input_files.emplace_back(
-          arg.str(), current_fshader_stage == shaderc_glsl_infer_from_source
-                         ? glslc::DeduceDefaultShaderKindFromFileName(arg)
-                         : current_fshader_stage);
+      input_files.emplace_back(InputFileSpec{
```
Comment (antiagainst) [inline glslc/src/main.cc]: I think we can directly place the three parameters in `emplace_back`?

File: glslc/README.asciidoc (hunk 1 above)
Comment (dneto0) [inline glslc/README.asciidoc]: Yeah, I wasn't sure.  :-0
Next commit tests that -std is ignored for HLSL, and documents it.

File: glslc/README.asciidoc (hunk 2 above)
Comment (dneto0) [inline glslc/README.asciidoc]: Done in next commits

**Comment by dneto0** [comment]
All feedback addressed.

File: glslc/src/main.cc (hunk 3 above)
Comment (antiagainst) [inline glslc/src/main.cc]: I see.

**Comment by dneto0** [comment]
Squashed and merged into master as cdefe18ac0

**Pull request closed without merging**